### ISSUE TITLE
Сделать приложение админки

### ISSUE THREAD

**Issue by TFK70** [atls-academy/Gradesor.github.io#3]
### Задача

Сделать приложение админки

 ### Роуты
 
 - `/` - должен происходить редирект на `/dashboard`
 - `/dashboard` - основная страница с панелью
 - `/auth` - страница авторизации

### Auth

На странице авторизации должно быть реализовано следующее:

- [ ] Форма авторизации
- [ ] Регистрация
- [ ] Логин
- [ ] Проверка, что в поле login нет кириллицы
- [ ] Проверка, что login.length > 3 символов
- [ ] Проверка, что password.length > 5 символов
- [ ] Проверка при регистрации, что пользователя с указанным логином не существует. Если существует - выбрасывать ошибку на фронт
- [ ] Обработка ошибок при логине: `User does not exist`, `Passwords do not match`
- [ ] Сохранение сессии в cookie
- [ ] Проверка, что в cookie нет активной сессии. Если есть - делать режирект на `/dashboard`
- [ ] После окончания флоу логина/регистрации делать редирект на  `/dashboard`

### Dashboard

**Требования по оформлению:**

На странице панели должен быть [holy grail](https://en.wikipedia.org/wiki/Holy_grail_(web_design))

- [ ] Header
- [ ] Content
- [ ] Left Sidebar
- [ ] Footer

**Функциональные требования:**

- [ ] Сделать роут защищенным. При попытке перейти на него без авторизации - делать редирект на `/auth`
- [ ] В `Header` должен отображаться логин авторизованного пользователя
- [ ] В `Header` должна быть кнопка `Logout`. При нажатии на нее сессия должна удалиться, должен произойти редирект на `/auth`
- [ ] В `Content` должно быть выведено 10 пользователей (users), расположенных в столбик. В итеме каждого пользователя должны отображаться `name`, `email` и аватарка. Эти свойства должны быть сгенерированы с помощью либы [faker](https://fakerjs.dev/) (там же можно получить url на рандомное изображение, которое будет использоваться в аватаре)
- [ ] `Left Sidebar` должен быть [Collapsed](https://www.w3schools.com/howto/tryit.asp?filename=tryhow_js_collapse_sidebar)
- [ ] В `Left Sidebar` должны быть контролсы для сортировки элементов: выбор, какой именно столбик сортируем и выбор по какому признаку будет производиться сортировка. Должна быть возможность сортировать столбик users по полям name и email, столбик news по полям date и content.
- [ ] В `Left Sidebar` должна быть кнопка сброса фильтров

### И самое главное

Весь написанный код должен быть готов к масштабированию. То есть при расширении модуля основная кодовая база не должна подвергаться мажорным изменениям

**Comment by Gradesor** [comment]
@TFK70  стек и сборку свою юзаю?)

**Comment by TorinAsakura** [comment]
> @TFK70 стек и сборку свою юзаю?)



> @TFK70 стек и сборку свою юзаю?) и есть ли диз?
> Если диза нет то можно использовать библиотеки компонентов? в т.ч. https://ant.design/docs/react/introduce
> Есть ли уже какой-то эндпоинт куда можно цепляться?

- дизайна нет
- компонентную базу либо бери из atls/uikit либо пиши сам
- требования к таске обновил

**Comment by TFK70** [comment]
> Есть ли уже какой-то эндпоинт куда можно цепляться?

Нет, флоу авторизации нужно реализовать на клиентской части, тут вообще бэка нету

**Comment by Gradesor** [comment]
@TFK70

1) это разве не одно и тоже?)
![image](https://user-images.githubusercontent.com/50404138/176756109-5df984e9-9561-4c56-a097-78e10d1b107e.png)
-----
2)
2.1)Пожелания к тексту ошибки есть?
2.2)На основе чего я могу понять зареган ли юзер с таким логином?
![image](https://user-images.githubusercontent.com/50404138/176756304-e1494249-a32e-41df-9c8b-76876f693707.png)
-----
3) что понимаешь под сессией?

**Comment by TFK70** [comment]
> это разве не одно и тоже?)

Под формой авторизации подразумевается сама морда, верстка, а под логином/регистрацией - функционал

> Пожелания к тексту ошибки есть?

Главное чтобы понятно было

> На основе чего я могу понять зареган ли юзер с таким логином?

Хранить массив на клиентской части с зареганными пользователями и сверяться с ним

> что понимаешь под сессией?

Когда пользователь проходит авторизацию ему выдается сессия. По ней можно идентифицировать пользователя (получить все данные о нем) и, соответственно, понять, авторизован пользователь или нет. Нет сессии === пользователь не авторизован

**Comment by Gradesor** [comment]
@TFK70 
> Хранить массив на клиентской части с зареганными пользователями и сверяться с ним

в indexedDb положу, ок?)

> Когда пользователь проходит авторизацию ему выдается сессия. По ней можно идентифицировать пользователя (получить все данные о нем) и, соответственно, понять, авторизован пользователь или нет. Нет сессии === пользователь не авторизован

Запариться с генерацией jwt? и его дешифровкой?

**Comment by Gradesor** [comment]
@TorinAsakura для обработки форм что-то ваше использовать? или могу свои приблуды?

**Comment by TorinAsakura** [comment]
@Gradesor Используй реакт

**Comment by Gradesor** [comment]
> @Gradesor Используй реакт


@TorinAsakura есть что почитать по тому куда и как вы декомпозируете и выносите логику?
Например: Валидация, сабмит формы

**Comment by TFK70** [comment]
> в indexedDb положу, ок?)

Да, тоже вариант

> Запариться с генерацией jwt? и его дешифровкой?

Нет

> просто светить ошибку или блокировать ввод?

Как угодно, главное чтобы флоу не проходил когда эти условия не выполнятся, но если ты блокируешь ввод - дай пользователю знать почему

> есть что почитать по тому куда и как вы декомпозируете и выносите логику?
сейчас интересует: Валидация, сабмит формы

Это сйечас на тебе, делай как умеешь
Но, если интересно, касательно формы авторизации на боевых проектах - её валидацией занимается кратос, на фронте мы выводим ошибки которые с него приходят

**Comment by Gradesor** [comment]
>Это сйечас на тебе, делай как умеешь
Но, если интересно, касательно формы авторизации на боевых проектах - её валидацией занимается кратос, на фронте мы выводим ошибки которые с него приходят

Ну я так и понял, что вы всё это дело через делаете`@atls/next-identity-integration`, который в свою очередь вяжется с кратосом))
